### ISSUE TITLE
ChangeSetInterception method name pluralization is backwards.

### ISSUE THREAD

**Issue by robertmclaws** [OData/RESTier#571]
In the course of developing Testier, I had to build a naming algorithm that would match the built-in convention system.

Consider the following list:
```
--------------------------------------------------
Function Name                            |   Found
--------------------------------------------------
CanInsertSports                          |   False
CanUpdateSports                          |   False
CanDeleteSports                          |   False
OnInsertingSports                        |   False
OnUpdatingSports                         |   False
OnDeletingSports                         |   False
OnFilterSport                            |    True
OnInsertedSports                         |    True
OnUpdatedSports                          |   False
OnDeletedSports                          |   False
```

Just browsing the list above, it starts to be come pretty clear. You don't "filter sport", you filter a list of sports, So those should be plural. You can't delete multiple sports from the before/after delete routines, so those should be singular.

Realizing that this will break everyone, it still needs to happen before RTM so that the methods are more intuitive.

**Comment by mikepizzo** [comment]
Making P1 since it will be a massive breaking change to try and take after RTM.

Seems like a pretty straightforward fix (and probably simply a bug in the if statement within the pluralization code that does exactly the wrong thing) -- Robert, any chance you can author a PR for this?

**Comment by robertmclaws** [comment]
Yeah, I'll take it. Should I branch off of master for the PR?

**Comment by robertmclaws** [comment]
@mikepizzo So, unfortunately it wasn't just a bug in the "if" code that picks names, as there is no centralized convention naming engine. [I did create one for Breakdance](https://github.com/advancedrei/Breakdance/blob/master/src/AdvancedREI.Breakdance.Restier/ConventionBasedMethodNameFactory.cs), however, and I'd be fine with integrating that and using it instead.

In the meantime, I've 
centralized convention naming engine. [I did create one for Breakdance](https://github.com/advancedrei/Breakdance/blob/master/src/AdvancedREI.Breakdance.Restier/ConventionBasedMethodNameFactory.cs), however, and I'd be fine with integrating that and using it instead.

In the meantime, I've pushed a fix. I added some neat code to give the user debug warnings if we found the old method names. We won't actually execute the old method names, but we will let them know they need to make a change, which is better than not having anything work and not knowing why.
